framework/src/ics/InitialConditionWarehouse.C

```diff
@@ -33,40 +33,45 @@ InitialConditionWarehouse::initialSetup(THREAD_ID tid)
 void
 InitialConditionWarehouse::addObject(MooseSharedPointer<InitialCondition> object, THREAD_ID tid)
 {
-  // Check that when object is boundary restricted that the variable is nodal
-  const MooseVariable & var = object->variable();
 
-  // Boundary Restricted
-  if (object->boundaryRestricted())
+  // Perform IC error checking, there is no need to do this on each thread, since they are all the same
```

Comment:
You could use a thread-safe container if it's feasible or add a thread
lock. While that might seem extreme this code only runs once so it's worth
the wait to be assured that the system is sane before trying to use it.

On Mon, Jan 4, 2016 at 11:40 AM Andrew E Slaughter <notifications@github.com>
wrote:

> In framework/src/ics/InitialConditionWarehouse.C
> <https://github.com/idaholab/moose/pull/6104#discussion_r48763308>:
>
> >
> > -  // Boundary Restricted
> > -  if (object->boundaryRestricted())
> > +  // Perform IC error checking, there is no need to do this on each thread, since they are all the same
>
> I could store tid copies of the name lists and do the checks on each
> thread, but that seems un-necessary
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/6104/files#r48763308>.
>